sharedstore/getter.go

```diff
@@ -48,6 +57,10 @@ type pollGetter struct {
 	store Store
 }
 
+func (g *pollGetter) WouldWait() bool {
+	return true
```

Comment:
And should this be similarly smarter?

Reply:
Added a check for the context, but I ultimately don't care about this one, the shared store should not give you a pollGetter if it's not supposed to wait anyway.